src/model/MutableClimbDataSource.ts

```diff
@@ -125,6 +125,23 @@ export default class MutableClimbDataSource extends ClimbDataSource {
         ? createGradeObject(grade, typeSafeDisciplines, cragGradeScales)
         : null
 
+      const pitches = userInput[i].pitches
+
+      const newPitchesWithIDs = pitches != null
+        ? pitches.map((pitch): IPitch => {
+          if (pitch.pitchNumber === undefined) {
+            throw new UserInputError('Each pitch in a multi-pitch climb must have a pitchNumber representing its sequence in the climb. Please ensure that every pitch is numbered.')
+          }
+
+          return {
+            ...pitch,
+            _id: muid.from(pitch.id ?? muid.v4()), // generate MUUID if not present
+            parentId: muid.from(pitch.parentId ?? newClimbIds[i]).toString(),
```

Comment:
Per my feedback above about changing IPitch.parentId to MUUID, we can just assign `newClimbIds[i]` to parentId (not doing toString())